src/device/intrinsics/math.jl

```diff
@@ -362,6 +362,14 @@ end
     ifelse(anynan, NaN, minval), ifelse(anynan, NaN, maxval)
 end
 
+@static if Base.thismajor(LLVM.version()) == v"18"
```

Comment:
Technically this is an issue with LLVM 14-18: https://godbolt.org/z/Pe3arzKcG. It just happens that with Julia <= v1.11we don't use methods which would trigger the issue, so we observe it only with LLVM 18 coupled to Julia v1.12.